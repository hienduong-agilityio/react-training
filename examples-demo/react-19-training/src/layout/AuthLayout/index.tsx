// Libraries
import { Outlet } from 'react-router-dom';

// Layouts
import { JSX } from 'react';

/**
 * AuthLayout component
 *
 * @returns {JSX.Element} The main layout structure with navigation components and content area.
 */
export const AuthLayout = (): JSX.Element => {
  return (
    <main className='flex flex-col justify-center items-center h-dvh bg-gray-400'>
      <Outlet />
    </main>
  );
};
