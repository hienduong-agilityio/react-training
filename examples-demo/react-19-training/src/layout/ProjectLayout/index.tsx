// Libraries
import { Outlet, useNavigate } from 'react-router-dom';

// SVG
// import backArrow from '@public/images/backArrow.svg';

// Constants
import { ROUTE } from '../../constants';
import { JSX } from 'react';

/**
 * Project layout component
 *
 * @returns {JSX.Element} - The Project layout element
 */
export const ProjectLayout = (): JSX.Element => {
  const navigate = useNavigate();

  const handleGoBack = () => {
    window.history.length > 1 ? navigate(-1) : navigate(ROUTE.PROJECT);
  };

  return (
    <div className='flex flex-col h-full bg-gray-300 p-6'>
      <button
        role='button'
        onClick={handleGoBack}
        className='mb-6 w-24 p-2 bg-white border-2 text-black flex gap-2 items-center justify-center focus:ring-4 focus:ring-primary-100 rounded-md cursor-pointer'
      >
        {/* <img src={backArrow} alt='Back Arrow' className='w-5 h-5 pr-2' /> */}
        Back
      </button>
      <div className='bg-white p-6 rounded-lg shadow-md w-full max-w h-auto'>
        <Outlet />
      </div>
    </div>
  );
};
