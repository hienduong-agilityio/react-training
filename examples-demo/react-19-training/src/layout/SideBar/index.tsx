// Libraries
import { Link } from 'react-router-dom';

// Constants
import { ROUTE } from '../../constants';

// Svg
// import logoIcon from '@public/images/logoIcon.svg';
// import projectsIcon from '@public/images/projectsIcon.svg';
// import settingIcon from '@public/images/settingIcon.svg';
import { JSX } from 'react';

/**
 * Sidebar components
 *
 * @returns {JSX.Element} The rendered Sidebar menu item.
 */
export const Sidebar = (): JSX.Element => {
  return (
    <div className='flex bg-primary-700'>
      <div className='w-[74px] bg-primary-700 h-screen p-4 relative duration-300'>
        <div className='flex items-center'>
          <Link to={ROUTE.PROJECT}>
            {/* <img src={logoIcon} className='cursor-pointer duration-500' alt='Logo' /> */}
          </Link>
        </div>
        <ul className='pt-10 flex flex-col items-center justify-center gap-5'>
          <li className='rounded-md items-center mt-2 focus:bg-primary-900 cursor-pointer hover:border-primary-100 hover:bg-primary-600'>
            <Link to={ROUTE.PROJECT}>
              {/* <img className='p-2' src={projectsIcon} alt='Projects' /> */}
            </Link>
          </li>
          <li className='rounded-md items-center mt-2 focus:bg-primary-900 cursor-pointer hover:border-primary-100 hover:bg-primary-600'>
            <Link to={ROUTE.ROOT}>
              {/* <img className='p-2' src={settingIcon} alt='Setting' /> */}
            </Link>
          </li>
        </ul>
      </div>
    </div>
  );
};
