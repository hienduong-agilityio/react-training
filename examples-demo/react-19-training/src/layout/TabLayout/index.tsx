// Libraries
// import { generateBreadcrumbItems } from '@/helpers/breadcrumb';
// import { useMemo } from 'react';
// import { useLocation, useParams } from 'react-router-dom';

import { JSX } from 'react';
import { ROUTE } from '../../constants';
import { Link } from 'react-router-dom';

// Components
// import { Breadcrumbs } from '@/components';

/**
 * TabLayout component
 *
 * @returns {JSX.Element} - The TabLayout element
 */
export const TabLayout = (): JSX.Element => {
  // const location = useLocation();
  // const { id: projectId } = useParams();

  // const breadcrumbItems = useMemo(
  //   () => generateBreadcrumbItems(location.pathname, projectId),
  //   [location.pathname, projectId]
  // );

  return (
    <div className='flex flex-col px-4 py-2'>
      {/* <Breadcrumbs items={breadcrumbItems} /> */}
      This breadcrumbs component
      <Link className='p-2 bg-primary-300' to={ROUTE.LOGIN}>
        Login
      </Link>
    </div>
  );
};
