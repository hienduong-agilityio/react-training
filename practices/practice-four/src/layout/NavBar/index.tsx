// Libraries
import { Link } from 'react-router-dom';

// Components
import { Avatar } from '@/components';

// Svg
import notifyIcon from '@public/images/notificationIcon.svg';
import helpIcon from '@public/images/helpIcon.svg';

// Constants
import { ROUTE } from '@/constants';

/**
 * Navbar components
 *
 * @returns {JSX.Element} The Navbar component.
 */
export const Navbar = (): JSX.Element => {
  return (
    <nav className='bg-light w-full border'>
      <div className='flex flex-wrap items-center justify-between p-4'>
        <Link to={ROUTE.PROJECT} className='text-2xl font-semibold text-gray-900'>
          Projects
        </Link>
        <div className='flex items-center gap-6'>
          <img src={notifyIcon} className='cursor-not-allowed' alt='Notification' />
          <img src={helpIcon} className='cursor-not-allowed' alt='Help' />
          <Avatar customClass='rounded-full cursor-not-allowed' name='HienDuong' />
        </div>
      </div>
    </nav>
  );
};
