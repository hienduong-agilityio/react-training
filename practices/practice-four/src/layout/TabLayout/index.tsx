// Libraries
import { generateBreadcrumbItems } from '@/helpers/breadcrumb';
import { useMemo } from 'react';
import { useLocation, useParams } from 'react-router-dom';

// Components
import { Breadcrumbs } from '@/components';

/**
 * TabLayout component
 *
 * @returns {JSX.Element} - The TabLayout element
 */
export const TabLayout = (): JSX.Element => {
  const location = useLocation();
  const { id: projectId } = useParams();

  const breadcrumbItems = useMemo(
    () => generateBreadcrumbItems(location.pathname, projectId),
    [location.pathname, projectId]
  );

  return (
    <div className='px-4 py-2'>
      <Breadcrumbs items={breadcrumbItems} />
    </div>
  );
};
