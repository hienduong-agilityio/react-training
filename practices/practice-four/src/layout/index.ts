// Index Layout files
export * from './MainLayout';
export * from './NavBar';
export * from './ProjectLayout';
export * from './SideBar';
export * from './TabLayout';
