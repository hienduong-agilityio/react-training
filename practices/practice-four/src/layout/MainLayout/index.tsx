// Libraries
import { Outlet } from 'react-router-dom';

// Layouts
import { Sidebar, Navbar, TabLayout } from '@/layout';

/**
 * MainLayout component
 *
 * @returns {JSX.Element} The main layout structure with navigation components and content area.
 */
export const MainLayout = (): JSX.Element => {
  return (
    <main className='flex'>
      <Sidebar />
      <div className='flex flex-col w-full'>
        <Navbar />
        <TabLayout />
        <Outlet />
      </div>
    </main>
  );
};
