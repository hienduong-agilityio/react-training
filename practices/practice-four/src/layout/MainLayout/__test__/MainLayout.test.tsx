// Libraries
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import { MainLayout } from '@/layout';

// Mock Layout
jest.mock('@/layout/Sidebar', () => ({
  Sidebar: () => <div>Mock Sidebar</div>
}));
jest.mock('@/layout/Navbar', () => ({
  Navbar: () => <div>Mock Navbar</div>
}));
jest.mock('@/layout/TabLayout', () => ({
  TabLayout: () => <div>Mock TabLayout</div>
}));

// Helper function to render MainLayout
const renderMainLayout = () => {
  const queryClient = new QueryClient();

  return render(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <MainLayout />
      </MemoryRouter>
    </QueryClientProvider>
  );
};

// Helper function to check layout components
const expectLayoutComponentsToBePresent = () => {
  expect(screen.getByText('Mock Sidebar')).toBeInTheDocument();
  expect(screen.getByText('Mock Navbar')).toBeInTheDocument();
  expect(screen.getByText('Mock TabLayout')).toBeInTheDocument();
};

/*
MainLayout component
  âˆš should render the sidebar, navbar and tabLayout
*/
describe('MainLayout', () => {
  it('should render the sidebar, navbar and tabLayout', () => {
    const { asFragment } = renderMainLayout();

    expectLayoutComponentsToBePresent();
    expect(asFragment()).toMatchSnapshot();
  });
});
