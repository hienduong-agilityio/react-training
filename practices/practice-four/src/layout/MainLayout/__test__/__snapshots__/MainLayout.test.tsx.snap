// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render the sidebar, navbar and tabLayout 1`] = `
<DocumentFragment>
  <main
    class="flex"
  >
    <div>
      Mock Sidebar
    </div>
    <div
      class="flex flex-col w-full"
    >
      <div>
        Mock Navbar
      </div>
      <div>
        Mock TabLayout
      </div>
    </div>
  </main>
</DocumentFragment>
`;
